File: src/main/java/com/github/sarhatabaot/messages/TypeKeyValue.java
package com.github.sarhatabaot.messages;

/**
 * @author sarhatabaot
 */
public class TypeKeyValue {
    private final Class<?> clazz;
    private final String value;

    public TypeKeyValue(final Class<?> clazz, final String value) {
        this.clazz = clazz;
        this.value = value;
    }
    
    /**
     * Returns the value to set.
     * @return the value.
     */
    public String getValue() {
        return value;
    }
    
    /**
     * Returns the class type to generate.
     * @return The class type.
     */
    public Class<?> getClazz() {
        return clazz;
    }
}

File: src/main/kotlin/com/github/sarhatabaot/messages/model/TypeKeyValue.kt
package com.github.sarhatabaot.messages.model

/**
 *
 * @author sarhatabaot
 */

class TypeKeyValue(private val clazz: Class<*>, private val value: String) {

    /**
     * Returns the value to set.
     * @return the value.
     */
    fun getValue(): String {
        return value
    }

    /**
     * Returns the class type to generate.
     * @return The class type.
     */
    fun getClazz(): Class<*> {
        return clazz
    }
}
